packages/mgt-element/src/components/baseComponent.ts

```diff
@@ -115,6 +118,34 @@ export abstract class MgtBaseComponent extends LitElement {
     if (this.isShadowRootDisabled()) {
       (this as any)._needsShimAdoptedStyleSheets = true;
     }
+    this.handleLocalizationChanged = this.handleLocalizationChanged.bind(this);
+  }
+
+  /**
+   * Invoked each time the custom element is appended into a document-connected element
+   *
+   * @memberof MgtBaseComponent
+   */
+  public connectedCallback() {
+    super.connectedCallback();
+    LocalizationHelper.onUpdated(this.handleLocalizationChanged);
+    this.updateDirection();
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    LocalizationHelper.removeOnUpdated(this.handleLocalizationChanged);
+  }
+
+  /**
+   * Request localization changes when the 'strings' event is detected
+   *
+   * @protected
+   * @memberof MgtBaseComponent
+   */
+  protected handleLocalizationChanged() {
+    this._strings = this.serveStrings(this._strings);
```

Comment:
You should add a protected strings getter here (so that components can overwrite it), and use that as the parameter to updateStringsForTag.

ex:

in baseComponent.ts:

```tsx
protected get strings() {
 return {};
}

protected handleLocalizationChanged() {
  this._strings = LocalizationHelper.updateStringsForTag(this.tagName, this.strings);
}
```

in a component:

```tsx

import { strings } from './strings';

export class MgtPerson... {
  protected get strings() {
    return strings;
  }
}

```